### PR TITLE
Polls push rules: synchronise poll rules with message rules

### PR DESCRIPTION
**Pull request by kerryarchibald**
Syncs poll-specific push rules from [MSC3930](https://github.com/matrix-org/matrix-spec-proposals/blob/travis/msc/polls-notifs/proposals/3930-polls-notifications.md) with message related push rules.

Will add cypress coverage separately https://github.com/vector-im/element-web/issues/24717

*`Notifications.tsx` has a lot of existing strict errors, won't fix in this PR

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Polls push rules: synchronise poll rules with message rules ([\#10263](https://github.com/matrix-org/matrix-react-sdk/pull/10263)). Contributed by @kerryarchibald.

### CODE REVIEW

File: src/components/views/settings/Notifications.tsx [hunk 1]
```diff
@@ -115,9 +126,65 @@ interface IState {
...
+ * Find the 'loudest' vector state assigned to a rule
+ * and it's synced rules
+ * If rules have fallen out of sync,
+ * the loudest rule can determine the display value
+ * @param defaultRules
+ * @param rule - parent rule
+ * @param definition - definition of parent rule
+ * @returns VectorState - the maximum/loudest state for the parent and synced rules
+ */
+const maximumVectorState = (
```
Comment (robintown) [inline src/components/views/settings/Notifications.tsx:155]: Some whitespace between these definitions would improve the readablity

File: src/components/views/settings/Notifications.tsx [hunk 2]
```diff
@@ -191,6 +260,7 @@ export default class Notifications extends React.PureComponent<IProps, IState> {
                 phase: Phase.Ready,
             });
         } catch (e) {
+            console.log(e);
             logger.error("Error setting up notifications for settings: ", e);
```
Comment (robintown) [inline src/components/views/settings/Notifications.tsx:264]: Aren't these log lines redundant?

File: src/components/views/settings/Notifications.tsx [hunk 3]
```diff
@@ -388,6 +459,48 @@ export default class Notifications extends React.PureComponent<IProps, IState> {
...
+     * Updated syncedRuleIds from rule definition
+     * If a rule does not exist it is ignored
+     * Synced rules are updated sequentially
+     * and stop at first error
+     */
+    private updateSyncedRules = async (
+        syncedRuleIds: VectorPushRuleDefinition["syncedRuleIds"],
+        actions?: PushRuleAction[],
+    ): Promise<void> => {
+        console.log(
```
Comment (robintown) [inline src/components/views/settings/Notifications.tsx:486]: To be consistent, this should go through `logger`

**Review by robintown — APPROVED** [review]

File: src/components/views/settings/Notifications.tsx (hunk 2 above)
Comment (kerryarchibald) [reply src/components/views/settings/Notifications.tsx:264]: Debugs I missed when tidying up, removed.

**Pull request merged**